lib/src/matchers.rs

```diff
@@ -19,18 +19,14 @@ use std::collections::{BTreeSet, HashMap, HashSet};
 use crate::repo_path::{RepoPath, RepoPathComponent};
 
 #[derive(PartialEq, Eq, Debug)]
-pub struct Visit {
-    dirs: VisitDirs,
-    files: VisitFiles,
-}
-
-impl Visit {
-    pub fn all() -> Self {
-        Self {
-            dirs: VisitDirs::All,
-            files: VisitFiles::All,
-        }
-    }
+pub enum Visit {
+    /// Everything in the directory is *guaranteed* to match, no need to check
+    /// descendants
+    AllRecursively,
+    Some {
```

Comment:
Is 'Some' a good name, given its normal connotations?

Reply:
Yes, makes sense to rename it. Went with "Specific".